tests/0126-oauthbearer_oidc.c

```diff
@@ -33,64 +33,140 @@
 
 
 /**
- * @brief After config OIDC fields, make sure the producer gets created
- *        successfully.
+ * @brief After config OIDC, make sure the producer and consumer
+ *        can work successfully.
  *
  */
-static void do_test_create_producer() {
+static void do_test_produce_consumer_with_OIDC(rd_kafka_conf_t *conf) {
+        if (test_needs_auth()) {
+                TEST_SKIP("Mock cluster does not support SSL/SASL\n");
+                return;
+        }
+
         const char *topic;
         uint64_t testid;
-        rd_kafka_t *rk;
-        rd_kafka_conf_t *conf;
-        rd_kafka_conf_res_t res;
-        char errstr[512];
+        rd_kafka_t *p1;
+        rd_kafka_t *c1;
+        int msgcounter;
+        rd_kafka_topic_t *rkt;
 
-        SUB_TEST("Test producer with oidc configuration");
+        SUB_TEST("Test producer and consumer with oidc configuration");
 
-        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "sasl.oauthbearer.token.endpoint.url",
+                      "https://dev-531534.okta.com/oauth2/default/v1/token");
 
-        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.method", "oidc", errstr,
-                                sizeof(errstr));
+        testid = test_id_generate();
 
-        if (res == RD_KAFKA_CONF_INVALID) {
-                rd_kafka_conf_destroy(conf);
-                TEST_SKIP("%s\n", errstr);
+        rd_kafka_conf_set_dr_msg_cb(rd_kafka_conf_dup(conf), test_dr_msg_cb);
+
+        p1 = test_create_handle(RD_KAFKA_PRODUCER, rd_kafka_conf_dup(conf));
+
+        topic = test_mk_topic_name("0126-oauthbearer_oidc", 1);
+        test_create_topic(p1, topic, 1, 3);
+        TEST_SAY("Topic: %s is created\n", topic);
+
+        rkt = test_create_producer_topic(p1, topic);
+        TEST_SAY("Auto-creating topic %s\n", topic);
+        test_auto_create_topic_rkt(p1, rkt, tmout_multip(5000));
+
+        test_produce_msgs_nowait(p1, rkt, testid, 0, 0, 1, NULL, 0, 0,
+                                 &msgcounter);
+        TEST_SAY("Flushing..\n");
+        rd_kafka_flush(p1, 1000);
```

Comment:
1s is pretty short, and you are not checking that the flush succeeded.
Use test_flush() with a timeout of 10 or so.

Reply:
But since you are flushing after nowait you can just as well just use `test_produce_msgs2()` which will perform the wait for you